node/api/client/renter.go

```diff
@@ -43,6 +44,46 @@ func (c *Client) RenterDownloadFullGet(siaPath, destination string, async bool)
 	return
 }
 
+// RenterClearAllDownloadsPost requests the /renter/downloads/clear resource
+// with no parameters
+func (c *Client) RenterClearAllDownloadsPost() (err error) {
+	values := url.Values{}
+	values.Set("start", "")
+	values.Set("end", "")
+	err = c.post("/renter/downloads/clear", values.Encode(), nil)
+	return
+}
+
+// RenterClearDownloadsAfterPost requests the /renter/downloads/clear resource
+// with only the end time provided
+func (c *Client) RenterClearDownloadsAfterPost(end time.Time) (err error) {
+	values := url.Values{}
+	values.Set("start", "")
+	values.Set("end", strconv.FormatInt(end.UnixNano(), 10))
+	err = c.post("/renter/downloads/clear", values.Encode(), nil)
+	return
+}
+
+// RenterClearDownloadsBeforePost requests the /renter/downloads/clear resource
+// with only the start time provided
+func (c *Client) RenterClearDownloadsBeforePost(start time.Time) (err error) {
+	values := url.Values{}
+	values.Set("start", strconv.FormatInt(start.UnixNano(), 10))
+	values.Set("end", "")
+	err = c.post("/renter/downloads/clear", values.Encode(), nil)
+	return
+}
+
+// RenterClearDownloadsRangePost requests the /renter/downloads/clear resource
+// with both start and end times provided
+func (c *Client) RenterClearDownloadsRangePost(start, end time.Time) (err error) {
+	values := url.Values{}
+	values.Set("start", strconv.FormatInt(start.UnixNano(), 10))
+	values.Set("end", strconv.FormatInt(end.UnixNano(), 10))
+	err = c.post("/renter/downloads/clear", values.Encode(), nil)
+	return
+}
```

Comment:
not sure if we need four separate functions. Calling `RenterClearDownloads(start, time.Time{})` doesn't seem so bad. Also, should it be `RenterDownloadsClear` to match the route?

Reply:
When only one value is submitted, how to you recognize if it is the start or end?

Reply:
I think I actually recommended using multiple functions since I thought it might improve usability for developers. Just feels a bit more intuitive to me. Unfortunately go doesn't have default parameters :/

Reply:
you can sort of fake default parameters:
```go
func RenterClearDownloads(args ...time.Time) {
    if len(args) == 1 {
        // assume args[0] is start time
    } else if len(args) == 2 {
        // assume args[0] is start time, args[1] is end time
    } else {
        // invalid
    }
}
```
but this is still suboptimal (e.g. when you only want to specify the end time) and it's easier to make a mistake.

Having four separate functions makes me sad, but I don't see a better alternative, so let's stick with it for now.